libmscore/layout.cpp

```diff
@@ -1339,6 +1343,12 @@ void Score::hideEmptyStaves(System* system, bool isFirstSystem)
                     && !(isFirstSystem && styleB(Sid::dontHideStavesInFirstSystem))
                     && hideMode != Staff::HideMode::NEVER)) {
                   bool hideStaff = true;
+                  for (auto spanner : spanners) {
+                        if (spanner.value->staff() == staff) {
```

Comment:
Simlpe fix seems to be to add ` && !spanner.value->systemFlag()` to the condition here.

Reply:
Good catch! I'll go ahead and make that tweak today, as well as add something to the test case to catch future regressions of that specific variety.

Reply:
On the danger to repeat myself: For getting regressions like this caught better and quicker, we'd really need 'nightly builds', so have the PRs on top of the 3.x or master branch.
Having them on top of 3.x and they'd easily could get `git cherry-pick`ed (I did the opposite and cherry-picxked them from 3.6.2_backend to 3,x, but I'm just one and can't kill'em all ;-)), having them on master (too) takes care of them not getting forgotten.